### PR TITLE
Tests migrate to new approach

### PR DESCRIPTION
**Pull request by wprzechrzta**

### CODE REVIEW

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/EnvoyControlSynchronizationTest.kt [hunk 1]
```diff
@@ -1,124 +1,155 @@
...
+            readinessCheck = { name, target -> waitServiceOkAndFrom(name, target) }
+        )
 
         // then
         logger.info("remote dc latency: $latency")
 
         val tolerance = Duration.ofMillis(400) + stateSampleDuration
         val expectedMax = (pollingInterval + tolerance).toMillis()
-        assertThat(latency.max()).isLessThanOrEqualTo(expectedMax)
+        Assertions.assertThat(latency.max()).isLessThanOrEqualTo(expectedMax)
```
Comment (Automaat) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/EnvoyControlSynchronizationTest.kt:94]: why did you removed static imports for `assertions` and `await`

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/consul/ConsulClusterExtension.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        consuls.forEach { consul ->
+            val consulContainerNames = consuls.map { it.container.containerName() }.toTypedArray()
+            val args = arrayOf("consul", "join", *consulContainerNames)
+            consul.container.execInContainer(*args)
+        }
+        started = true
+    }
+
+    override fun afterAll(context: ExtensionContext?) {
```
Comment (Automaat) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/consul/ConsulClusterExtension.kt:35]: do we need that `AfterAllCallback` if it doesn't do anything?

**Review by Automaat — DISMISSED** [review]

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/consul/ConsulMultiClusterExtension.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.junit.jupiter.api.extension.AfterAllCallback
+import org.junit.jupiter.api.extension.AfterEachCallback
+import org.junit.jupiter.api.extension.BeforeAllCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.testcontainers.containers.Network
+
+class ConsulMultiClusterExtension : BeforeAllCallback, AfterAllCallback, AfterEachCallback {
+
+    companion object {
+        private val FIRST_CLUSTER = ConsulClusterSetup(listOf(
```
Comment (lukidzi) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/consul/ConsulMultiClusterExtension.kt:12]: Can't we reuse `ConsulClusterExtension`?

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsLoggingModeTest.kt [hunk 4]
```diff
@@ -80,668 +85,683 @@ internal class IncomingPermissionsLoggingModeTest : EnvoyControlTestConfiguratio
...
-
-    private fun callPost(service: String, from: EnvoyContainer, path: String) = call(
-        service = service, from = from, path = path, method = "POST",
-        body = RequestBody.create(MediaType.get("text/plain"), "{}"))
-
     private val EnvoyContainer.ingressSslRequests: Int?
-    get() = this.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()
+        get() = this.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()
 
     private val EnvoyContainer.ingressPlainHttpRequests: Int?
```
Comment (lukidzi) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsLoggingModeTest.kt:765]: Maybe lets replace this code with direct calls in test `envoy().container.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()` (like in other tests)

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/EnvoyControlSynchronizationTest.kt (hunk 1 above)
Comment (wprzechrzta) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/EnvoyControlSynchronizationTest.kt:94]: by accident, will correct  it

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/consul/ConsulClusterExtension.kt (hunk 2 above)
Comment (wprzechrzta) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/consul/ConsulClusterExtension.kt:35]: nop

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsLoggingModeTest.kt (hunk 4 above)
Comment (wprzechrzta) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsLoggingModeTest.kt:765]: ok

**Review by Automaat — APPROVED** [review]

**Review by lukidzi — APPROVED** [review]

**Pull request merged**